### ISSUE TITLE
Fragments are not being processed when included in another fragment

### ISSUE THREAD

**Issue by kop** [async-graphql/async-graphql#345]
I believe it's a major bug that can lead to a lot of unpredictable issues on GraphQL clients.

## Steps to Reproduce the Problem

1. `git clone --recurse-submodules git@github.com:async-graphql/async-graphql.git`
2. `cd examples && cargo run --bin actix-web-starwars`
3. In GraphQL Playground, execute the following query:

```graphql
query {
  ...FragmentA
  humans {
    edges {
      node {
        id
      }
    }
  }
}

fragment FragmentA on QueryRoot {
  ...FragmentB
  ...FragmentC
}

fragment FragmentB on QueryRoot {
  humans {
    edges {
      node {
        name
      }
    }
  }
}

fragment FragmentC on QueryRoot {
  humans {
    edges {
      node {
        homePlanet
      }
    }
  }
}
```

## Expected Behavior

Returned should be:

<details><summary>Click me</summary>
<p>

```json
{
  "data": {
    "humans": {
      "edges": [
        {
          "node": {
            "homePlanet": null,
            "id": "1004",
            "name": "Wilhuff Tarkin"
          }
        },
        {
          "node": {
            "homePlanet": "Tatooine",
            "id": "1001",
            "name": "Luke Skywalker"
          }
        },
        {
          "node": {
            "homePlanet": "Tatooine",
            "id": "1000",
            "name": "Luke Skywalker"
          }
        },
        {
          "node": {
            "homePlanet": "Alderaa",
            "id": "1003",
            "name": "Leia Organa"
          }
        },
        {
          "node": {
            "homePlanet": null,
            "id": "1002",
            "name": "Han Solo"
          }
        }
      ]
    }
  }
}
```

</p>
</details>

## Actual Behavior

Returned data is:

<details><summary>Click me</summary>
<p>

```json
{
  "data": {
    "humans": {
      "edges": [
        {
          "node": {
            "id": "1004"
          }
        },
        {
          "node": {
            "id": "1001"
          }
        },
        {
          "node": {
            "id": "1000"
          }
        },
        {
          "node": {
            "id": "1003"
          }
        },
        {
          "node": {
            "id": "1002"
          }
        }
      ]
    }
  }
}
```

</p>
</details>

## Specifications

- Version: 2.1.4
- Platform: N/A
- Subsystem: N/A

**Comment by sunli829** [comment]
Thank you for reporting this issue and I will resolve it as soon as possible. 微笑

**Comment by sunli829** [comment]
Fixed in `v2.1.5`. 🙂

**Comment by kop** [comment]
Thank you for such a fast fix! Everything works now.
